### PR TITLE
IEditAndContinueSolutionProvider

### PR DESCRIPTION
**Pull request by tmat**
Implements https://github.com/dotnet/roslyn/issues/55683

Contributes to a fix of https://github.com/dotnet/roslyn/issues/55761

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue
+{
+    /// <summary>
+    /// Retrieves the <see cref="Solution"/> snapshot that corresponds to the current state of the debuggee.
+    /// This snapshot contains changes successfully applied during EnC/Hot Reload.
+    /// </summary>
+    /// <remarks>
+    /// This is temporarily available to in-proc XAML External Access APIs and should be revisited once XAML moves to OOP LSP.
+    /// </remarks>
+    internal interface IEditAndContinueSolutionProvider
```
Comment (spadapet) [inline src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs:14]: >internal [](http://example.com/codeflow?start=4&length=8)

This should be public

Comment (mgoertz-msft) [reply src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs:14]: If this is only temporary, then you don't want to make this public and therefore available to anyone. You can consume this from our XAML assembly in Roslyn and expose another internal interface from there that we can see from XamlLanguageService in VS via IVT.

File: src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        Solution? CommittedSolution { get; }
```
Comment (spadapet) [inline src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs:16]: >Solution? CommittedSolution { get; } [](http://example.com/codeflow?start=8&length=36)

I think I will need an event for when CommittedSolution changes. Any other event I use for when hot reload happens may not fire at the right time.

File: src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs (hunk 1 above)
Comment (spadapet) [reply src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs:14]: OK I didn't know about a XAML assembly in Roslyn. That works as long as all the internals are visible to the right assemblies.

Comment (mgoertz-msft) [reply src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs:14]: It's only visible to one assembly, Microsoft.VisualStudio.DesignTools.XamlLanguageService, but that should be enough.

**Pull request closed without merging**